vllm/engine/llm_engine.py

```diff
@@ -950,14 +1000,20 @@ def _process_model_outputs(self,
                 finished_before.append(i)
                 continue
 
-            if len(outputs) > 1:
+            if has_multi_step_outputs:
                 output = outputs_by_sequence_group[i]
+                assert len(output) == \
+                            self.scheduler_config.num_scheduler_steps, \
+                    f"#outputs {len(output)}, #scheduler-steps {self.scheduler_config.num_scheduler_steps}" # noqa
             else:
                 output = [outputs_by_sequence_group[0][i]]
 
-            if not is_async:
-                seq_group.update_num_computed_tokens(
-                    scheduled_seq_group.token_chunk_size)
+            # TODO (varun) : What is the significance of `not is_async here`?
```

Comment:
Why is the update only required when `not is_async` is used ?

Reply:
I understand now - `is_async` actually refers to the processing of the outputs of step N  at the next step 0 (where is `--num-scheduler-steps`). Ill remove the TODO.